app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java

```diff
@@ -330,7 +333,19 @@ public OkHttpJsonApiClient(OkHttpClient okHttpClient,
                 for (MwQueryPage page : pages) {
                     Media media = Media.from(page);
                     if (media != null) {
-                        mediaList.add(media);
+                        if(keyword.equals("Category:Uploaded_with_Mobile/Android")){
+
+                            String wikiText = "";
+                            try {
+                                wikiText = mediaWikiApi.getWikiText(media.getPageTitle());
+                            } catch (Exception e){
+                                e.printStackTrace();
+                            }
+
+                            if(!wikiText.contains("{{delete")){
+                                mediaList.add(media);
+                            }
+                        } else mediaList.add(media);
```

Comment:
I would suggest putting that into its own
```
{
    ...
}
```
block, to improve visibility, even if not required by the Java language.

Per https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used (which is linked to from https://github.com/commons-app/apps-android-commons/wiki/Contributing-Guidelines )